x-pack/plugins/security_solution/public/detections/components/host_isolation/crowdstrike_agent_status.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiBadge, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { getAgentStatusText } from '../../../common/components/endpoint/agent_status_text';
+import { HOST_STATUS_TO_BADGE_COLOR } from '../../../management/pages/endpoint_hosts/view/host_constants';
+import {
+  ISOLATED_LABEL,
+  ISOLATING_LABEL,
+  RELEASING_LABEL,
+} from '../../../common/components/endpoint/endpoint_agent_status';
+import { useAgentStatusHook } from './use_external_host_isolation';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+
+const EuiFlexGroupStyled = styled(EuiFlexGroup)`
+  .isolation-status {
+    margin-left: ${({ theme }) => theme.eui.euiSizeS};
+  }
+`;
+
+export const CrowdstrikeAgentStatus = React.memo(
+  ({ agentId, 'data-test-subj': dataTestSubj }: { agentId: string; 'data-test-subj'?: string }) => {
+    const useAgentStatus = useAgentStatusHook();
+
+    const crowdstrikeManualHostActionsEnabled = useIsExperimentalFeatureEnabled(
+      'responseActionsCrowdstrikeManualHostIsolationEnabled'
+    );
+
+    const { data, isLoading, isFetched } = useAgentStatus([agentId], 'crowdstrike', {
+      enabled: crowdstrikeManualHostActionsEnabled,
+    });
+    const agentStatus = data?.[`${agentId}`];
```

Comment:
Nit: `agentId` is declared as a string, so this templating/wrapping/casting is unnecessary:

```suggestion
    const agentStatus = data?.[agentId];
```